src/lib/spinel/radio_spinel.cpp

```diff
@@ -1584,7 +1584,7 @@ void RadioSpinel::HandleTransmitDone(uint32_t          aCommand,
 
     static_cast<Mac::TxFrame *>(mTransmitFrame)->SetIsHeaderUpdated(headerUpdated);
 
-    if ((sRadioCaps & OT_RADIO_CAPS_TRANSMIT_SEC) && headerUpdated &&
+    if ((sRadioCaps & OT_RADIO_CAPS_TRANSMIT_SEC) && (!mIsHeaderUpdatedBeforeSending) && headerUpdated &&
```

Comment:
we don't need mIsHeaderUpdatedBeforeSending

Reply:
Updated